cmake/HPX_AddConfigTest.cmake

```diff
@@ -424,285 +242,6 @@ function(hpx_check_for_cxx11_std_atomic_128bit)
-endfunction()
-
-###############################################################################
-function(hpx_check_for_cxx11_std_initializer_list)
-  add_hpx_config_test(HPX_WITH_CXX11_STD_INITIALIZER_LIST
-    SOURCE cmake/tests/cxx11_std_initializer_list.cpp
-    FILE ${ARGN})
-endfunction()
-
-###############################################################################
-function(hpx_check_for_cxx11_std_is_bind_expression)
-  add_hpx_config_test(HPX_WITH_CXX11_IS_BIND_EXPRESSION
-    SOURCE cmake/tests/cxx11_std_is_bind_expression.cpp
-    FILE ${ARGN})
-endfunction()
-
-###############################################################################
-function(hpx_check_for_cxx11_std_is_placeholder)
-  add_hpx_config_test(HPX_WITH_CXX11_IS_PLACEHOLDER
-    SOURCE cmake/tests/cxx11_std_is_placeholder.cpp
-    FILE ${ARGN})
-endfunction()
-
-###############################################################################
-function(hpx_check_for_cxx11_std_is_trivially_copyable)
-  add_hpx_config_test(HPX_WITH_CXX11_IS_TRIVIALLY_COPYABLE
-    SOURCE cmake/tests/cxx11_std_is_trivially_copyable.cpp
-    FILE ${ARGN})
-endfunction()
-
-###############################################################################
-function(hpx_check_for_cxx11_std_lock_guard)
-  add_hpx_config_test(HPX_WITH_CXX11_LOCK_GUARD
-    SOURCE cmake/tests/cxx11_std_lock_guard.cpp
-    FILE ${ARGN})
-endfunction()
-
-###############################################################################
-function(hpx_check_for_cxx11_std_random)
-  add_hpx_config_test(HPX_WITH_CXX11_RANDOM
-    SOURCE cmake/tests/cxx11_std_random.cpp
-    FILE ${ARGN})
-endfunction()
-
-###############################################################################
-function(hpx_check_for_cxx11_std_range_access)
-  add_hpx_config_test(HPX_WITH_CXX11_RANGE_ACCESS
-    SOURCE cmake/tests/cxx11_std_range_access.cpp
-    FILE ${ARGN})
-endfunction()
-
-###############################################################################
-function(hpx_check_for_cxx11_std_reference_wrapper)
-  add_hpx_config_test(HPX_WITH_CXX11_REFERENCE_WRAPPER
-    SOURCE cmake/tests/cxx11_std_reference_wrapper.cpp
-    FILE ${ARGN})
-endfunction()
-
-###############################################################################
-function(hpx_check_for_cxx11_std_regex)
-  add_hpx_config_test(HPX_WITH_CXX11_REGEX
-    SOURCE cmake/tests/cxx11_std_regex.cpp
-    FILE ${ARGN})
-endfunction()
-
-###############################################################################
-function(hpx_check_for_cxx11_std_shared_ptr)
-  add_hpx_config_test(HPX_WITH_CXX11_SHARED_PTR
-    SOURCE cmake/tests/cxx11_std_shared_ptr.cpp
-    FILE ${ARGN})
-endfunction()
-
-###############################################################################
-function(hpx_check_for_cxx11_std_shared_ptr_lwg3018)
-  add_hpx_config_test(HPX_WITH_CXX11_SHARED_PTR_LWG3018
-    SOURCE cmake/tests/cxx11_std_shared_ptr_lwg3018.cpp
-    FILE ${ARGN})
-endfunction()
```

Comment:
Are we sure that this is really implemented by all compilers by now?

Reply:
No, I can add this one back. I can't find any reliable information on when it's been fixed so better keep it.

Reply:
I added this one and the `quick_exit` check back.